<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
	<title>Genome Annotation • Generally Biology</title>
    <meta name="description" content="ENTER THE BEST">
    <meta name="keywords" content="">
    
    	<!-- Twitter Cards -->
	<meta name="twitter:title" content="Genome Annotation">
	<meta name="twitter:description" content="ENTER THE BEST">
	
	
	
	<meta name="twitter:card" content="summary_large_image">
	<meta name="twitter:image" content="/images/forest.jpg">
	
	<!-- Open Graph -->
	<meta property="og:locale" content="en">
	<meta property="og:type" content="article">
	<meta property="og:title" content="Genome Annotation">
	<meta property="og:description" content="ENTER THE BEST">
	<meta property="og:url" content="/bioinformatics/GenomeAnnotation/">
	<meta property="og:site_name" content="Generally Biology">

    <link rel="canonical" href="/bioinformatics/GenomeAnnotation/">

    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Generally Biology Atom Feed">
    <link href="/sitemap.xml" type="application/xml" rel="sitemap" title="Sitemap">

    <meta name="HandheldFriendly" content="True">
    <meta name="MobileOptimized" content="320">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="cleartype" content="on">

    <link rel="stylesheet" href="/css/main.css">
    <!-- HTML5 Shiv and Media Query Support for IE -->
    <!--[if lt IE 9]>
      <script src="/js/vendor/html5shiv.min.js"></script>
      <script src="/js/vendor/respond.min.js"></script>
    <![endif]-->

  </head>

  <body id="js-body">
    <!--[if lt IE 9]><div class="upgrade notice-warning"><strong>Your browser is quite old!</strong> Why not <a href="http://whatbrowser.org/">upgrade to a newer one</a> to better enjoy this site?</div><![endif]-->

    <header id="masthead">
  <div class="inner-wrap">
    <a href="/" class="site-title">Generally Biology</a>
    <nav role="navigation" class="menu top-menu">
        <ul class="menu-item">
	<li class="home"><a href="/">Generally Biology</a></li>
	
    
    <li><a href="/" >Home</a></li>
  
    
    <li><a href="/blog/" >Blog</a></li>
  
    
    <li><a href="/collections/" >Collections</a></li>
  
    
    <li><a href="/about/" >About</a></li>
  
</ul>
    </nav>
  </div><!-- /.inner-wrap -->
</header><!-- /.masthead -->

    <nav role="navigation" id="js-menu" class="sliding-menu-content">
  <h5>Generally Biology <span>Table of Contents</span></h5>
  <ul class="menu-item">
    <li>
      <a href="/">
        
        <div class="title">Home</div>
        <p class="excerpt">Generally Biology</p>
      </a>
    </li><li>
      <a href="/blog/">
        
        <div class="title">Blog</div>
        <p class="excerpt">My Writings on Biology, and computers, and the insane intersection between the two that we call bioinformatics/computational biology.</p>
      </a>
    </li><li>
      <a href="/collections/">
        
        <div class="title">Collections</div>
        <p class="excerpt">Writings on specific subjects</p>
      </a>
    </li><li>
      <a href="/about/">
        
        <div class="title">About</div>
        <p class="excerpt">John Pablo Mendieta</p>
      </a>
    </li>
  </ul>
</nav>
<button type="button" id="js-menu-trigger" class="sliding-menu-button lines-button x2" role="button" aria-label="Toggle Navigation">
  <span class="nav-lines"></span>
</button>

<div id="js-menu-screen" class="menu-screen"></div>


    <div id="page-wrapper">
      <div id="main" role="main">
	<article class="wrap" itemscope itemtype="http://schema.org/Article">
		
		<div class="page-feature">
			<div class="page-image">
				<img src="/images/forest.jpg" class="page-feature-image" alt="Genome Annotation" itemprop="image">
				
			</div><!-- /.page-image -->
		</div><!-- /.page-feature -->
		
		
  <nav class="breadcrumbs">
    <span itemscope itemtype="http://data-vocabulary.org/Breadcrumb">
      <a href="" itemprop="url">
        <span itemprop="title">Home</span>
      </a> › 
    <span itemscope itemtype="http://data-vocabulary.org/Breadcrumb">
      <a href="/Bioinformatics/" itemprop="url">
        <span itemprop="title">Bioinformatics</span>
      </a>
    </span>
  </nav><!-- /.breadcrumbs -->

		<div class="page-title">
			<h1>Genome Annotation</h1>
		</div>
		<div class="inner-wrap">
			<div id="content" class="page-content" itemprop="articleBody">
				<p>The human genome may have been sequenced in 2001. Im the time between 2001 and
2017 genome sequencing adn assembly have become common place. There are a host
of well annotated software suites that can take you from raw sequence to
genomes in a short amount of time and explain every step to you. However,
i hate to tell you that this is nothing like genomic annotation.</p>

<p>When I sarted annotating the Bitter Melon genome summer 2017 I was startled to
learn that there were almost no reources explaining how to take an assembled
genome from scaffolds to gene annoations. So, with that realization I created
this guide in order to distill everything I learned from this process to
hopefully make your experiance a little more tranparerent than mine was.</p>

<p>But Before we begin the annotation process I would encourage you to be honest
with the resources that you’re currently bringing to the table for this
annotation. Well annotated genomes require alot of resources, and if you simply
don’t hvae the resources than this pursiot may simply not be worth it. Terrible genome annotations are worse than useless.</p>

<p>This guide will assume though that you have the following things at your disposal. If you’re missing one of these items it may not be cripaling but it will probably negativly affect your genome annotation.</p>

<ol>
  <li>Well assembled genome (Scaffold size &gt; mean gene size)</li>
  <li>RNA seq data, the more the better</li>
  <li>Closely related species that are decently annotated</li>
  <li>Large amount of computational resources</li>
</ol>

<p>In my experience MAKER tends to yield the best genome annotations. So, much of the data prodcued will be fed into maker. Instead of giving maker any raw data we will instead be feeding in all the data we process ourselves. This is for two fold reasons. 1. Maker tends to spit out pretty poor assemblies when you feed it raw data, and 2. We need to ensure that out intermediates are the highest quality we can. Genome annotation is already rife with errors no matter how well done, so by synthesizing these parts seperatly we can hopefully catch erros that might have been glossed over in a larger pipeline such as maker.</p>

<p>##1 Repeat Annotation</p>

<p>In order to annotate you genome with any semblance of correctness we have to
first create a repeat library with which to mask the genome. IN order to do
this I utalized RepeatMasker, RepeatScout, REPark, ltrdigest, and MITEhunter.</p>

<p>I loosly followed the MAKER turtiruial in creating a basic repeat library. I
atempted to run use the advanced repeat libraray sections but a large majority
of the code was broken.</p>

<p>If interested my scripts to generate all this stuff can be found at mu github
here. The scripts are failry simple bash scripts and will require a bit of
editing, but are far simpler than some of the other turtorials you will find
online.</p>

<p>But what you’re going to need to create at the end of this step is a repeat
libary file containing both MITE, LTRs, and any sequence identified by Repark.
We will use thils file far further down the pipeline.</p>

<ul>
  <li>Quick note** *</li>
</ul>

<p>Mite hunter is a horrible error laiden piece of software that hasn’t been
updated or maintained since 2007. This simple little package took me 2 days of
trouble shooting to finally get working at one point. At soem point I might
have to write a spereate guide on creating painless repeatlibraries utalizing
docker and nextflow (if i ever have enough time for that)</p>

<p>##2 Trinity assemblies</p>

<p>In order to utilize RNA seq we need to create three different transcriptomes
assmeblies for pasa and MAKER to use. We are going to need a DeNovo genome
assembly, a genome guided assembly and finally read alignmenst using HISAT or
cufflinks in order to generate a .bam file. It’s important to gauge just how
many BUSCOs are within each transcriptome assembly</p>

<p>##Pasa Pipeline</p>

<p>To create the best training set we possibly we can we are going to create a
agapped transcriptome alignment. To do some i used PASA pipeline produced by
Brian Haas of the broad institure. I’m using pasa because based off it’s paper
where it was compared to similar programs (genemark, etc…) it per formaed the
best with the most accuracy.</p>

<p>Installing pasa can be rather intimidating if you haven’t installed mysql
before. But, it’s far simpler than many online blogs would have you believe.
SInce mysql has so many variable flavors the documentation is rather
nightmarish when you attempt to untangle what version or flavor to install.
So, to install pasa pipelien i ran.</p>

<div class="highlighter-rouge"><pre class="highlight"><code></code></pre>
</div>

<p>##SPALN Gapped alignments</p>

<p>Next, we need to create a gapped protein alignemnt. Since w</p>

<h2 id="augutust">Augutust</h2>

<p>This is one of the most challenging parts of genomic annotation. Since we
cannot rely on gapped alignments of prior genomes to predict novel genes, we
are forced to implement software which utalizes hidden markov models. These
models, while effective, are incredibly challening to train and can take many
iterations to achieve detection levels sensative and accuarate enough to
identify novel genes.</p>

<p>The gene predictor i will be training is Augustus. AUgustus has become the
software of choice for many gene predcictors. However, while heavily used,
there are very poor resources out there illustrating how to create an effective
and accurate training set. After much struggle and a little guidance from other
bioinformaticians here at BTI i was able to create a training set that
performed fairly well on a novel genome.</p>

<p>So, in order to train augustus we need to isolate a set of training genes that
we are certain are represenative of the genome as a whole. To do this i created
a nice litte pipeline in python which utalizes closely related species, and
recipricol blasting to generate a high quality training set. After using this
pipeline i was able to get a training set that had a sensativity of .857, which
I think is pretty good. So below i Will illustrate how i created this training
set in more detail.</p>


				<hr />
				<footer class="page-footer">
					

<div class="author-image">
	<img src="/images/" alt="">
</div><!-- ./author-image -->
<div class="author-content">
	<h3 class="author-name" >Written by Pablo Mendieta <span itemprop="author"></span></h3>
	<p class="author-bio"></p>
</div><!-- ./author-content -->

					<div class="inline-btn">
	<a class="btn-social twitter" href="https://twitter.com/intent/tweet?text=Genome%20Annotation&amp;url=/bioinformatics/GenomeAnnotation/&amp;via=" target="_blank"><i class="fa fa-twitter" aria-hidden="true"></i>  Twitter</a>
	<a class="btn-social facebook" href="https://www.facebook.com/sharer/sharer.php?u=/bioinformatics/GenomeAnnotation/" target="_blank"><i class="fa fa-facebook" aria-hidden="true"></i>  Facebook</a>

	</div><!-- /.share-this -->

					<div class="page-meta">
	<p>Updated <time datetime="2017-07-07T11:00:00Z" itemprop="datePublished">July 07, 2017</time></p>
</div><!-- /.page-meta -->

				</footer><!-- /.footer -->
				<aside>
					<hr />
<div id="disqus_thread"></div>
<script type="text/javascript">
	/* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
	var disqus_shortname = '';

	/* * * DON'T EDIT BELOW THIS LINE * * */
	(function() {
		var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
		dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
		(document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
	})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

				</aside>
			</div><!-- /.content -->
		</div><!-- /.inner-wrap -->
		
	</article><!-- ./wrap -->
</div><!-- /#main -->

      <footer role="contentinfo" id="site-footer">
	<nav role="navigation" class="menu bottom-menu">
		<ul class="menu-item">
		
      
			<li><a href="" >Follow:</a></li>
		
      
			<li><a href="https://twitter.com/pabster212?lang=en" target="_blank">Twitter</a></li>
		
      
			<li><a href="https://github.com/Jome0169" target="_blank">Github</a></li>
		
      
			<li><a href="https://www.linkedin.com/in/john-mendieta-016305119/" target="_blank">Linkedin</a></li>
		
		</ul>
	</nav><!-- /.bottom-menu -->
	<p class="copyright">&#169; 2017 <a href="">Generally Biology</a> powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> + <a href="http://mmistakes.github.io/skinny-bones-jekyll/" rel="nofollow">Skinny Bones</a>.</p>
</footer>

    </div>

    <script src="/js/vendor/jquery-1.9.1.min.js"></script>
    <script src="/js/main.js"></script>

  </body>

</html>
